modular_skyrat/modules/reagent_forging/code/crafting_bench.dm

```diff
@@ -0,0 +1,242 @@
+	if(goal_item_path)
+		clear_required()
+		balloon_alert_to_viewers("table cleared!")
+		update_appearance()
+		return
+	var/target_choice = tgui_input_list(user, "Which item would you like to craft?", "Crafting Choice", allowed_choices)
+	if(!target_choice)
+		balloon_alert(user, "no choice made!")
+		return
+	goal_name = target_choice
+	goal_item_path = allowed_choices[target_choice]
+	switch(target_choice)
+		if("Chain Helmet")
+			required_chain = 5
+		if("Chain Armor")
+			required_chain = 6
+		if("Chain Gloves")
+			required_chain = 4
+		if("Chain Boots")
+			required_chain = 4
+		if("Plated Boots")
+			required_plate = 4
+		if("Horseshoes")
+			required_chain = 4
+		if("Ring")
+			required_chain = 2
+		if("Collar")
+			required_chain = 3
+		if("Handcuffs")
+			required_chain = 10
+		if("Borer Cage")
+			required_plate = 6
+		if("Pavise Shield")
+			required_plate = 8
+		if("Buckler Shield")
+			required_plate = 5
+		if("Coil")
+			required_chain = 2
+	if(!required_hits)
+		required_hits = (required_chain * 2) + (required_plate * 2) + (required_coil * 2)
+	balloon_alert(user, "choice made!")
+	update_appearance()
+
+/obj/structure/reagent_crafting_bench/proc/clear_required()
+	required_hits = 0
+	current_hits = 0
+	goal_item_path = null
+	goal_name = null
+	required_chain = 0
+	required_plate = 0
+	required_coil = 0
+
+/obj/structure/reagent_crafting_bench/proc/check_required_materials()
+	if(chain_amount < required_chain)
+		return FALSE
+	if(plate_amount < required_plate)
+		return FALSE
+	if(coil_amount < required_coil)
+		return FALSE
+	return TRUE
+
+//when inserting the materials
+/obj/structure/reagent_crafting_bench/attackby(obj/item/attacking_item, mob/user, params)
+	update_appearance()
+	//if we are attacking with a hammer and we have a goal in mind!
+	if(istype(attacking_item, /obj/item/forging/hammer))
+		playsound(src, 'modular_skyrat/modules/reagent_forging/sound/forge.ogg', 50, TRUE)
+		if(length(contents))
+			if(istype(contents[1], /obj/item/forging/complete))
+				var/obj/item/forging/complete/complete_content = contents[1]
+				if(!complete_content?.spawning_item)
+					balloon_alert(user, "no craftable!")
+					return
+				if(current_wood < 2)
+					balloon_alert(user, "not enough wood!")
+					return
+				current_wood -= 2
+				var/spawning_item = complete_content.spawning_item
+				qdel(complete_content)
+				new spawning_item(src)
+				user.mind.adjust_experience(/datum/skill/smithing, 15) //creating grants you something
+				balloon_alert(user, "item crafted!")
+				update_appearance()
+				return
+		if(!goal_item_path)
+			balloon_alert(user, "no choice made!")
+			return
+		if(!check_required_materials())
+			balloon_alert(user, "not enough materials!")
+			return
+		var/skill_modifier = user.mind.get_skill_modifier(/datum/skill/smithing, SKILL_SPEED_MODIFIER) * 1 SECONDS
+		if(!COOLDOWN_FINISHED(src, hit_cooldown))
+			current_hits -= 3
+			balloon_alert(user, "bad hit!")
+			if(current_hits <= -required_hits)
```

Comment:
do we want required_hits to be negative here? or am i misunderstanding this line completely and the - is something else